python/perf-kernels/rmsnorm.py

```diff
@@ -0,0 +1,213 @@
+        triton.Config({'waves_per_eu': 2}, num_warps=8, num_stages=1),
+        triton.Config({'waves_per_eu': 2}, num_warps=16, num_stages=1),
+        triton.Config({'waves_per_eu': 4}, num_warps=4, num_stages=1),
+        triton.Config({'waves_per_eu': 4}, num_warps=8, num_stages=1),
+        triton.Config({'waves_per_eu': 4}, num_warps=16, num_stages=1),
+    ]
+
+def get_autotune_config():
+    if is_cuda():
+        return get_cuda_autotune_config()
+    else:
+        return get_hip_autotune_config()
+
+@triton.autotune(
+    configs=get_autotune_config(),
+    key=['n_rows', 'n_cols'],
+    use_cuda_graph=True
+)
+@triton.jit
+def rms_kernel(output_ptr, input_ptr, g_ptr, input_row_stride, output_row_stride, n_rows, n_cols, epsilon, BLOCK_SIZE: tl.constexpr):
+    row_start = tl.program_id(0)
+    row_step = tl.num_programs(0)
+    for row_idx in tl.range(row_start, n_rows, row_step):
+        row_start_ptr = input_ptr + row_idx * input_row_stride
+        col_offsets = tl.arange(0, BLOCK_SIZE)
+        mask = col_offsets < n_cols
+        
+        row = tl.load(row_start_ptr + col_offsets, mask=mask, other=0.0)
+        g = tl.load(g_ptr + col_offsets, mask=mask, other=0.0)
+        row_norm = row * row #square each value
+        row_norm = tl.sum(row_norm, axis=-1) #sum across columns(axis=-1)
+        row_norm = row_norm / n_cols #divide by n_cols
+        row_norm = row_norm + epsilon #add epsilon
+        row_norm = tl.sqrt(row_norm) #take sqrt, this is normalization value
+        rms_norm = row / row_norm #divide each x by normalization value
+        rms_norm = rms_norm * g #element wise multiplication with g
+        
+        output_row_start_ptr = output_ptr + row_idx * output_row_stride
+        output_ptrs = output_row_start_ptr + col_offsets
+        tl.store(output_ptrs, rms_norm, mask=mask)
+
+def rmsnorm(x, epsilon=1e-6):
+    n_rows, n_cols = x.shape
+    BLOCK_SIZE = triton.next_power_of_2(n_cols)
+
+    y = torch.empty_like(x,device='cuda')
+    g = torch.ones((1, n_cols), device='cuda')
+
+    num_programs = n_rows
+    grid = lambda meta: (num_programs,)
+    rms_kernel[grid](y,
+                    x,
+                    g,
+                    x.stride(0),
+                    y.stride(0),
+                    n_rows,
+                    n_cols,
+                    epsilon,
+                    BLOCK_SIZE
+    )
+
+    return y
+
+def run_rmsnorm(M, N):
+    torch.manual_seed(0)
+    x = torch.randn(M, N, device='cuda')
+    #print(x)
+    y_triton = rmsnorm(x)
+    #print(y_triton)
+
+    return y_triton
+
+
+@pytest.mark.parametrize('M, N', [
+    (1,4),
+    (2,10),
+    (8192,4096),
+    (4096,8192),
+    (1,8192),
+    (873,1245),
+])
+def test_rmsnorm(M, N):
+    torch.manual_seed(0)
+    x = torch.randn(M, N, device='cuda')
+    #print(x)
+    y_triton = rmsnorm(x)
+    #print(y_triton)
+
+    rms_norm = torch.nn.RMSNorm(N, device='cuda')
+    y_torch = rms_norm(x)
+    #print(y_torch)
```

Comment:
Can we remove these debug print statements before merging?

Reply:
Fixed